bld/unit_testers/build-namelist_test.pl

```diff
@@ -924,303 +839,270 @@ sub cat_and_create_namelistinfile {
      "bgc_laistreams_input"     =>{ options=>"--envxml_dir . --bgc bgc",
                                      namelst=>"stream_year_first_lai=1999",
-                                     GLC_TWO_WAY_COUPLING=>"FALSE",
                                      phys=>"clm5_0",
                                      },
      "crop_laistreams_input"     =>{ options=>"--envxml_dir . --bgc sp --crop",
                                      namelst=>"use_lai_streams=.true.",
-                                     GLC_TWO_WAY_COUPLING=>"FALSE",
                                      phys=>"clm5_0",
                                      },
      "soil_erod_wo_Zender"      =>{ options=>"--envxml_dir . --ignore_warnings",
                                      namelst=>"dust_emis_method='Leung_2023', " .
                                      "stream_meshfile_zendersoilerod = '/dev/null'",
-                                     GLC_TWO_WAY_COUPLING=>"FALSE",
                                      phys=>"clm5_1",
                                      },
      "soil_erod_wo_lnd_source"  =>{ options=>"--envxml_dir .",
                                      namelst=>"dust_emis_method='Zender_2003', " .
                                      "stream_fldfilename_zendersoilerod = '/dev/null', zender_soil_erod_source='atm'",
+                                     phys=>"clm5_1",
+                                     },
+     "soil_erod_none_w_Zender"  =>{ options=>"--envxml_dir .",
+                                     namelst=>"dust_emis_method='Zender_2003', " .
+                                     "zender_soil_erod_source='none'",
+                                     phys=>"clm5_1",
+                                     },
+     "soil_erod_bad_w_Zender"   =>{ options=>"--envxml_dir .",
+                                     namelst=>"dust_emis_method='Zender_2003', " .
+                                     "zender_soil_erod_source='zztop'",
+                                     phys=>"clm5_1",
+                                     },
+     "Set_Dust_When_CAM_Sets"   =>{ options=>"--envxml_dir .",
+                                     namelst=>"dust_emis_method='Zender_2003'",
+                                     LND_SETS_DUST_EMIS_DRV_FLDS=>"FALSE",
+                                     phys=>"clm5_1",
+                                     },
+     "soil_erod_none_w_Zender"  =>{ options=>"--envxml_dir .",
+                                     namelst=>"dust_emis_method='Zender_2003', " .
+                                     "zender_soil_erod_source='none'",
+                                     GLC_TWO_WAY_COUPLING=>"FALSE",
+                                     phys=>"clm5_1",
+                                     },
+     "soil_erod_bad_w_Zender"   =>{ options=>"--envxml_dir .",
+                                     namelst=>"dust_emis_method='Zender_2003', " .
+                                     "zender_soil_erod_source='zztop'",
```

Comment:
Also, is it ok that most (all?) dust-related namelst lines have a comma "," right after the method and end with a period "."? Looks inconsistent to me relative to other examples.

Reply:
You are right about the duplication. Thanks for pointing that out. I think I might have added it on the previous branch that came in as well as this one. So I'll remove the second duplicate one.

Reply:
Good question on the ending with the period. The syntax is correct, but as you point out it looks very arcane. The thing that's happening is that the "." concatenates two strings together. So the lines here...

``` perl
dust_emis_method='Zender_2003', " .
                                     "zender_soil_erod_source='none'",
```

Is really this...

``` perl
dust_emis_method='Zender_2003', zender_soil_erod_source='none'",
```

I did the concatenate to break up the long line. But, this might be a case that goes along with our recent discussion about long lines in code. With modern editors there isn't a reason to break lines up as much as in the past. The NAG compiler for FORTRAN code has a 132 character limit. So there's still reason to break up our FORTRAN code. And for readability I'd argue you should keep FORTRAN code shorter than that. But, perl and python don't have these limits. It's still an ingrained habit for me to shorten code lines, but this is a case where the concatenation makes the code unreaable. So I'll work with and removing that confusing use of concatenation syntax.

Reply:
OK, handled both these comments.